src/combobox/ComboBox.ts

```diff
@@ -0,0 +1,426 @@
+		const {
+			autoBlur,
+			onChange
+		} = this.properties;
+
+		this.updateSelectedIndex(Operation.reset);
+		this._open = false;
+		this._focused = !autoBlur;
+		this._ignoreFocus = true;
+		onChange && onChange(this._getResultLabel(result));
+	}
+
+	private _skipResult() {
+		this.updateSelectedIndex(this._direction);
+		this.invalidate();
+	}
+
+	@onPropertiesChanged
+	protected onPropertiesChanged(evt: PropertiesChangeEvent<this, ComboBoxProperties>) {
+		if (!this.registry || includes(evt.changedPropertyKeys, 'customResultItem') || includes(evt.changedPropertyKeys, 'customResultMenu')) {
+			this.registry = this._createRegistry();
+		}
+	}
+
+	protected scrollIntoView(element: HTMLElement) {
+		const menu = <HTMLElement> element.parentElement;
+		// Scroll menu up so top of highlighted result aligns with top of menu container
+		if (element.offsetTop - menu.scrollTop < 0) {
+			menu.scrollTop = element.offsetTop;
+		}
+		// Scroll menu down so bottom of highlighted result aligns with bottom of menu container
+		else if ((element.offsetTop - menu.scrollTop + element.offsetHeight) > menu.clientHeight) {
+			menu.scrollTop = element.offsetTop - menu.clientHeight + element.offsetHeight;
+		}
+	}
+
+	protected updateSelectedIndex(operation?: Operation, index?: number) {
+		const { results } = this.properties;
+
+		if (!results || results.length === 0 || !this._open) {
+			return;
+		}
+
+		if (index !== undefined) {
+			this._selectedIndex = index;
+		}
+		else if (operation !== undefined) {
+			if (operation === Operation.increase || operation === Operation.decrease) {
+				const total = results.length;
+				const base = operation === Operation.increase ? 0 : -1;
+				const current = this._selectedIndex !== undefined ? this._selectedIndex + operation : base;
+				this._selectedIndex = (current + total) % total;
+			}
+			else {
+				this._selectedIndex = undefined;
+			}
+			this._direction = operation;
+		}
+	}
+
+	render(): DNode {
+		const {
+			clearable,
+			disabled,
+			required,
+			invalid,
+			formId,
+			label,
+			readOnly,
+			results = [],
+			inputProperties = {},
+			value = ''
+		} = this.properties;
+
+		const menu = this._renderMenu(results);
+
+		this._notifyMenuChange();
+
+		this._wasOpen = this._open;
+
+		const container = v('div', {
+			bind: this,
+			classes: this.classes(css.combobox),
+			afterCreate: this._afterCreate,
+			afterUpdate: this._afterUpdate
+		}, [
+			// Use TextInput once landed
+			w(TextInput, {
+				bind: this,
+				classes: this.classes(clearable ? css.clearable : null),
+				disabled,
+				readOnly,
+				invalid,
+				required,
+				onBlur: this._onInputBlur,
+				onFocus: this._onInputFocus,
+				onInput: this._onInput,
+				onKeyDown: this._onInputKeyDown,
+				value,
+				overrideClasses: css,
+				...inputProperties
+			}),
+			clearable ? v('button', {
+				bind: this,
+				classes: this.classes(css.clear),
+				innerHTML: 'clear combo box',
+				onclick: this._onClearClick
```

Comment:
The button should also have `aria-controls` set to the id of the dropdown. You could optionally add a tabindex of -1, since the dropdown is accessible with keyboard interaction and the button isn't strictly necessary. It's also fine to leave it in the focus order, so it's up to you.